app/models/publication.rb

```diff
@@ -449,6 +449,10 @@ def has_open_access_information?
     !preferred_open_access_url.blank? || scholarsphere_upload_pending? || open_access_waived?
   end
 
+  def orcid_allowed?
+    doi.present? || url.present? || open_access_url.present?
+  end
```

Comment:
Does this reflect a requirement on orcid's end? Or does it just not make sense to create works that don't have any of this information?

In any case, we should take a closer look at these fields to be sure that we're actually getting the data that we want.

Reply:
Just a note for our meeting:  From the responses I'm getting from the Orcid API and sifting through their docs, it appears they require at least one "external-id".  However, it gets more complicated than that: that one required external-id must have an "external-id-relationship" of "self".  The value of "external-id-relationship" is a bit arbitrary and up to the client, but in their documentation they don't classify isbns or issns or any other location that does not point you directly to the article as "self"; they are classified as "part-of".  So, what I've gathered from this is that a work must have at least a doi or url as an external-id.  That raises the question: Do we want all of the urls/dois uploaded, or only some?  And which one? (:url, :open_access_url, :user_submitted_open_access_url)